shared/lib/sharing/publicize/actions.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	FETCH_PUBLICIZE_CONNECTIONS,
+	RECEIVE_PUBLICIZE_CONNECTIONS,
+	FAIL_PUBLICIZE_CONNECTIONS_REQUEST
+} from 'lib/redux-store/action-types';
+
+/**
+ * Triggers a network request to fetch Publicize connections for the specified
+ * site ID.
+ *
+ * @param  {Number}   siteId Site ID
+ * @return {Function}        Action thunk
+ */
+export function fetchConnections( siteId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: FETCH_PUBLICIZE_CONNECTIONS,
+			siteId
+		} );
+
+		return new Promise( ( resolve ) => {
```

Comment:
FWIW I think the approach you implemented is a good one and works well. Perhaps my initial understanding of Redux was slightly incorrect, because I don't see anything explicitly stating this shouldn't work like this now.